src/Libraries/Microsoft.Extensions.Telemetry.Testing/Metering/MetricCollector.cs

```diff
@@ -136,8 +137,9 @@ public void Dispose()
             // wake up anybody still waiting and inform them of the bad news: their horse is dead...
             foreach (var w in _waiters)
             {
-                w.TaskSource.SetException(MakeObjectDisposedException());
+                _ = w.TaskSource.TrySetException(MakeObjectDisposedException());
```

Comment:
I'm not sure if `ObjectDisposedException` is the right thing to do here. Setting the TCS to canceled might be more idiomatic. I'd need to look at what other APIs do for pending async tasks on dispose.

Reply:
I decided not to use cancel semantics here since I think that would be surprising. The caller didn't cancel anything, they disposed an object.

Reply:
I looked up what another type does here: `HttpClient` cancels any outstanding requests instead of throwing ODE - https://github.com/dotnet/runtime/blob/05e25ff8c3a623df74720f430c270eba0f1d41b0/src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs#L706-L710

I don't care much, but I'm pretty sure most .NET APIs clean up pending tasks on dispose by canceling the tasks, not throwing ODE.